presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -182,7 +186,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         }
 
         return Optional.of(new LimitApplicationResult<>(
```

Comment:
Do we have a tests that covers cases like `filter(limit(scan)` and `limit(filter(scan))`. In other words I think we should consider banning filter pushdown if there is already limit applied as I guess it is buggy today.

Reply:
That's something I wanted to ask. It seems we can push down either a filter or a limit, but not both.
I debugged it and found that the framework is doing the right thing. When a filter is present it does not push the limit down (i.e. does not call applyLimit). So that works fine, and I assume the "framework" has tests for that.

Reply:
Framework might have such tests but for regular tables. I think we need to add new ones for `information_schema` tables.